src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -171,7 +171,7 @@ public abstract partial class Diagnostic : IEquatable<Diagnostic?>, IFormattable
         /// <param name="severity">The diagnostic's effective severity.</param>
         /// <param name="defaultSeverity">The diagnostic's default severity.</param>
         /// <param name="isEnabledByDefault">True if the diagnostic is enabled by default</param>
-        /// <param name="warningLevel">The warning level, between 1 and 4 if severity is <see cref="DiagnosticSeverity.Warning"/>; otherwise 0.</param>
+        /// <param name="warningLevel">The warning level. For <see cref="DiagnosticSeverity.Error"/> severity this must be 0. For other <see cref="DiagnosticSeverity"/> severity levels set between 1 and 4.</param>
```

Comment:
I think it's a bit misleading. Any warning level includes errors, so "must be 0 for error" is misleading.